pkg/controller/applicationsets/controller.go

```diff
@@ -198,9 +207,15 @@ func (e *external) Delete(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalDelete{}, errors.New(errNotApplicationSet)
 	}
 
-	_, err := e.client.Delete(ctx, &applicationset.ApplicationSetDeleteRequest{
+	query := &applicationset.ApplicationSetDeleteRequest{
 		Name: meta.GetExternalName(cr),
-	})
+	}
+
+	if cr.Spec.ForProvider.AppsetNamespace != "" {
+		query.AppsetNamespace = cr.Spec.ForProvider.AppsetNamespace
+	}
```

Comment:
Do we need an extra check here? I think we can just set the value in the query directly because the default value is `""` anyways.

Same goes for the other controller.